### DISCUSSION TITLE
Any fork of Fingerprintjs' open source version that is more accurate?

### DISCUSSION THREAD

**Discussion by mohaa7** [fingerprintjs/fingerprintjs#833] [General]
The %60 accuracy of the opensource fingerprint does not meet my needs!
Do know any fork or work on increasing the accuracy of it?

**Comment by makma** [comment]
Hello @mohaa7,
I'm not aware of any more accurate open-source fingerprinting library than FingerprintJS. Needless to say, for Fingerprint Pro, we do offer quite a generous Free tier https://fingerprint.com/pricing/.

**Reply by Finesse** [reply]
I'd like to add the the open-source version accuracy is limited by the stability of the entropy sources. Fingerprint Pro doesn't have stable sources that aren't used in the open-source version. Fingerprint Pro uses tricky algorithms on the server side to perform a fuzzy matching of visits. The open-source versions is fully client-side, so can't compare the current visit with the other visits unless you download the whole visit database to the client side.

**Reply by mohaa7** [reply]
+1
All Pro improvements are server-side?

**Reply by Finesse** [reply]
Additional browser data that can’t be used to make an identifier on the client side, and server-side improvements

**Comment by mohaa7** [comment]
I'm wondering why no fuzzy matching on what was calculated client-side by `fingerprintjs` itself? I mean also fuzzy matching on the JSON generated by fingerprintjs, then no need for client-side calculation for a fingerprint!

**Reply by makma** [reply]
There are many reasons:

- Size and speed of such an implementation on the client side.
- Questionable reliability and accuracy considering just JS context without a reliable shared persistent layer.
- Immunity against tampering.
- Signals that can't be obtained just on the client side.
- Server-side validations because the result fingerprint might be easily counterfeited by a malicious party.
- Economic scalability - open source free tool requiring tens of engineers working full-time, conducting research, maintaining the project, adding features... for free

**Reply by Finesse** [reply]
For fuzzy matching you need the fingerprint components of all users that have visited your website. The FingerprintJS library provides a set of components for a single user (for obvious reasons).

You can do fuzzy matching on the server side after collecting the fingerprint components of several users. This task is very different computer science challenge (data science), so it's outside of the scope of the library.

**Comment by perpi** [comment]
Hmm, I got curious. That is, you save a hash (fingerprint) from FingerprintJS, and also generate a hash from machine learning on other data that you bring to your server. Well, what can be done with these two hashes?! Now, what does comparing two hashes mean to increase the certainty of the user's uniqueness?

**Reply by Finesse** [reply]
No hashes. Something like:

```ts
function getVisitorId(
  currentComponents: Record<string, unknown>,
  visitHistory: { visitorId: string, components: Record<string, unknown> }[],
): string {
  // Data science
  // Save the current components and the visitor id to the database
}
```

Where `visitHistory` is a database of thousands of identifications. Visitor ids returned by Fingerprint Pro are random string. I can't tell the exact data science.

Hashes are only necessary for FingerprintJS Open-Source to convert a map of components to a small string.
